cgo/walletloader.go

```diff
@@ -220,6 +226,8 @@ func closeWallet(cName *C.char) *C.char {
 	if err := w.CloseWallet(); err != nil {
 		return errCResponse("close wallet %q error: %v", name, err.Error())
 	}
+	close(w.shutdown)
+	w.Wait()
```

Comment:
If rescan is in progress when `w.CloseWallet()` is called, I'm not sure the rescan will be stopped before closing the wallet db? For sync for example, we call `w.StopSync()` and `w.WaitForSyncToStop()` in `w.CloseWallet` before closing the wallet db.

Might be a good idea to do something similar for rescans. One approach would be to create a cancellable context in `w.RescanProgressFromHeight` and cancel the context when `w.CloseWallet()` is called. Would also need a way to ensure that the rescan has actually stopped before closing the wallet db.

Reply:
I thought closing the wallet db would just cause functions trying to use it to error?

so, close w.shutdown and wait before CloseWallet and dont wait for ctx close in rescan?

Reply:
I'm pretty sure decrediton and probably cryptopower allow rescanning while sync is active. But maybe not. Anyway I think this solves the other comments https://github.com/decred/libwallet/compare/a28d2cf76b1853ea81ca31b04847e786b512d1bf..7419307d7d5185ec785928157b296390dbde47d5

Reply:
> I thought closing the wallet db would just cause functions trying to use it to error?

Eh. No idea. I suppose it could result in a panic but I absolutely don't know. Can leave as is, if that's okay by you.

Reply:
Closing w.shutdown before CloseWallet